api/src/opentrons/protocol_engine/state/addressable_areas.py

```diff
@@ -462,6 +470,30 @@ def get_addressable_area_center(self, addressable_area_name: str) -> Point:
             z=position.z,
         )
 
+    def get_fixture_by_deck_slot_name(
+        self, slot_name: DeckSlotName
+    ) -> Optional[CutoutFixture]:
+        """Get the Cutout Fixture of a fixture currently loaded into a specific Deck Slot."""
+        deck_config = self.state.deck_configuration
+        if deck_config:
+            slot_cutout_id = DECK_SLOT_TO_CUTOUT_MAP[slot_name]
+            slot_cutout_fixture = None
+            # This will only ever be one under current assumptions
+            for cutout_id, cutout_fixture_id in deck_config:
+                if cutout_id == slot_cutout_id:
+                    slot_cutout_fixture = (
+                        deck_configuration_provider.get_cutout_fixture(
+                            cutout_fixture_id, self.state.deck_definition
+                        )
+                    )
+                    break
+            if slot_cutout_fixture is None:
+                raise CutoutDoesNotExistError(
+                    f"No Cutout was found in the Deck that matched provided slot {slot_name}."
+                )
+            return slot_cutout_fixture
+        return None
```

Comment:
Your call, but I think you can simplify this control flow a little bit by `return`ing where you're currently `break`ing. Then you don't have to keep track of the `None` value for `slot_cutout_fixture`.

```suggestion
        if deck_config:
            slot_cutout_id = DECK_SLOT_TO_CUTOUT_MAP[slot_name]
            # This will only ever be one under current assumptions
            for cutout_id, cutout_fixture_id in deck_config:
                if cutout_id == slot_cutout_id:
                    slot_cutout_fixture = (
                        deck_configuration_provider.get_cutout_fixture(
                            cutout_fixture_id, self.state.deck_definition
                        )
                    )
                    return slot_cutout_fixture
            raise CutoutDoesNotExistError(
                f"No Cutout was found in the Deck that matched provided slot {slot_name}."
            )
        return None
```